### DISCUSSION TITLE
Has the `$config` name been reserved in Nuxt3?

### DISCUSSION THREAD

**Discussion by mertsincan** [nuxt/nuxt#27645] [General]
Hello to Nuxt Team from PrimeVue,

First of all, thank you very much for this wonderful project. You are all doing such excellent work :) An interesting issue is currently being reported regarding Nuxt and PrimeVue. https://github.com/primefaces/primevue/issues/5895
As far as I understand, the `$config` name has been reserved by Nuxt. When I try to use my own variable named $config in any component outside of PrimeVue, Nuxt's $config object consistently appears instead.

https://nuxt.com/docs/guide/going-further/runtime-config#exposing

> Public runtime config is accessible in Vue templates with $config.public.

Test;
```
// in my app
<template>
    <div>...</div>
</template>
<script>
export default {
    name: 'MyComponent',
    beforeCreate() {
        console.log(this.$config); // output: { public: ..., app: ...}
    },
    computed: {
        $config() {
            return { test: 'TEST'};
        }
    }
};
</script>
```

What do you think about this?

Best Regards,

**Comment by danielroe** [comment]
The $config name has actually been present since Nuxt v3 initial release, and I think even in Nuxt 2. So I guess this has always been a bug of some kind - it probably always returned undefined when you attempted to access primevue config from it?

It was only when we started logging invalid access to it that this became evident, I believe.

**Reply by Tarabass** [reply]
So where should this be solved? I believe $config is a common name and should be more specific for Nuxt, like $nuxtConfig. I also have the feeling that this could have a big impact on Nuxt.

For now I downgraded Nuxt to 3.11.2 to suppress the warnings.

**Reply by danielroe** [reply]
I believe this has already been resolved in the latest primevue RC.

**Reply by mertsincan** [reply]
Thanks a lot, @danielroe! Yes, we changed the $config name to $primevueConfig in PrimeVue 4.0-rc.2. This warning is gone. I just wanted to verify if this was a solution you suggested.

> it probably always returned undefined when you attempted to access primevue config from it?
- I'm not sure about the output on older PrimeVue versions, but I think it works fine. Because we use $config a lot and no errors have been reported by users.

Note: I created a dummy component without PrimeVue. When I tried it, I encountered the same warnings. I hope this helps you.

**Comment by lmc3s3f0r** [comment]
@mertsincan Is the same fix planned for PrimeVue 3.52 users?

**Reply by mertsincan** [reply]
Yes, fixed in https://github.com/primefaces/primevue/issues/5907 :)
